app/routes.py

```diff
@@ -1,2 +1,304 @@
+        task.description = request_body["description"]
+        
+
+        # save action and return what you created
+        db.session.add(task)
+        db.session.commit()
+        return {"task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete" : (False
+            if task.completed_at == None else True)
+            }},200
+    # delete task
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return {"details":f'Task {task.task_id} "{task.title}" successfully deleted'}
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods = ["PATCH"])
+def complete_task(task_id):
+    task = Task.query.get(task_id)
+    
+    # return error message if trying to complete task that doesnt exist
+    if task is None:
+            return make_response("", 404)
+    
+    
+    if request.method == "PATCH":
+        
+        # wave 4 slack bot, since we are using GET and it has to do with completing task, put here
+        task.completed_at = datetime.now()
+        slack_url = 'https://slack.com/api/chat.postMessage'
+        slack_params = {
+            "channel": "slack-api-test-channel",
+            "text": f"Someone just completed the task {task.title}"
+        }
+        slack_header = {
+            "Authorization": f"Bearer {os.environ.get('SLACK_API_TOKEN')}"
+        }
+        requests.get(url = slack_url, params = slack_params, headers = slack_header)
+
+        # going forward with PPATCh method
+        db.session.commit()
+        # hard coding the task to be true because this is complete task
+        return {"task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete" : True
+            }}
+    
+@tasks_bp.route("/<task_id>/mark_incomplete", methods = ["PATCH"])
+def incomplete_task(task_id):
+    task = Task.query.get(task_id)
+    
+    # if trying to mark task that doesnt exist, return none
+    if task is None:
+            return make_response("", 404)
+    
+    # hard coding the task to be none because this is incomplete task
+    if request.method == "PATCH":
+        task.completed_at = None 
+        db.session.commit()
+        return {"task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete" : False
+            }}
+    
+    
+@goals_bp.route("",methods=["POST", "GET"])
+def handle_goals():
+    
+    # will be very similar to handle_tasks just change variable names and delete somefields
+    
+    # get response body inJSON
+    if request.method == "POST":
+        request_body = request.get_json()
+
+        if 'title' not in request_body:
+            response_body = {
+                'details': 'Invalid data'
+            }
+            return make_response(jsonify(response_body), 400)
```

Comment:
No need to convert dictionaries into JSON, Flask will do that automatically!

```suggestion
            return response_body, 400
```